webapp/components/Comment.vue

```diff
@@ -140,24 +141,33 @@ export default {
       this.openEditCommentMenu = showMenu
       this.setEditPending(showMenu)
     },
-    async deleteCommentCallback() {
-      try {
-        var gqlMutation = gql`
-          mutation($id: ID!) {
-            DeleteComment(id: $id) {
-              id
-            }
-          }
-        `
-        await this.$apollo.mutate({
-          mutation: gqlMutation,
+    deleteCommentCallback() {
+      this.$apollo
+        .mutate({
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: store => {
+            const data = store.readQuery({
+              query: PostQuery(this.$i18n),
+              variables: { id: this.post.id },
+            })
+
+            const index = data.Post[0].comments.findIndex(
+              deletedComment => deletedComment.id === this.comment.id,
+            )
+            if (index !== -1) {
+              data.Post[0].comments.splice(index, 1)
+            }
+            store.writeQuery({ query: PostQuery(this.$i18n), data })
+          },
+        })
+        .then(res => {
+          this.$toast.success(this.$t(`delete.comment.success`))
+          this.$emit('deleteComment')
+        })
+        .catch(err => {
+          this.$toast.error(err.message)
```

Comment:
@Tirokk, I originally tried using `update` with `async/await`, but I couldn't the syntax correct.
I just changed it to using a promise cause I have examples in the code base of it working... we could refactor it, but I'm not sure if there is a need, just to let you know why I refactored it so.

Reply:
Just add `async` to the outer method, put a `await` before the request and wrap that inside the try/catch